src/main/java/net/rubygrapefruit/platform/internal/jni/DefaultOsxFileEventFunctions.java

```diff
@@ -0,0 +1,73 @@
+package net.rubygrapefruit.platform.internal.jni;
+
+import net.rubygrapefruit.platform.NativeException;
+import net.rubygrapefruit.platform.NativeIntegration;
+import net.rubygrapefruit.platform.internal.FunctionResult;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Queue;
+import java.util.Set;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+public class DefaultOsxFileEventFunctions implements NativeIntegration {
+
+    public boolean addRecursiveWatch(String path) {
```

Comment:
We should add some synchronization and lifecycling to prevent things like adding watches from multiple threads at the same time, or adding watches while we're already watching, or starting watching while we're already watching, and so on.

Reply:
👍